### PR TITLE
SSL/TLSize all the things! (convert http:// to https:// where possible)

### PR DESCRIPTION
**Pull request by reedloden**
In places where https:// is supported, let's use it. There shouldn't be any need for http:// at all, and it helps prevent people from being MITM'd.

I don't have a local testing environment, so we'll see if tests pass (and there may be linting issues), but I believe I covered most everything.

### CODE REVIEW

File: app/views/pages/download.html.erb [hunk 1]
```diff
@@ -6,10 +6,10 @@
     RubyGems is a package management framework for Ruby. Download the latest version here:
   </p>
   <div id="formats" class="download__formats">
-    <%= link_to "tgz", "http://production.cf.rubygems.org/rubygems/rubygems-#{version_number}.tgz", class: "download__format" %>
-    <%= link_to "zip", "http://production.cf.rubygems.org/rubygems/rubygems-#{version_number}.zip", class: "download__format" %>
-    <%= link_to "gem", "http://production.cf.rubygems.org/rubygems/rubygems-update-#{version_number}.gem", class: "download__format" %>
-    <%= link_to "git", "http://github.com/rubygems/rubygems", class: "download__format" %>
+    <%= link_to "tgz", "https://rubygems.global.ssl.fastly.net/rubygems/rubygems-#{version_number}.tgz", class: "download__format" %>
+    <%= link_to "zip", "https://rubygems.global.ssl.fastly.net/rubygems/rubygems-#{version_number}.zip", class: "download__format" %>
+    <%= link_to "gem", "https://rubygems.global.ssl.fastly.net/rubygems/rubygems-update-#{version_number}.gem", class: "download__format" %>
```
Comment (dwradcliffe) [inline app/views/pages/download.html.erb]: :+1:

**Comment by konklone** [comment]
Nice job!

File: app/views/pages/download.html.erb (hunk 1 above)
Comment (sferik) [inline app/views/pages/download.html.erb]: I’m :+1: for using HTTPS everywhere but I’m concerned about hard-coding domains that we don’t own. Is that a concern of anyone else? I’m thinking about people who copy these URLs into their scripts. I would rather have canonical HTTPS rubygems.org URLs that 302 to Fastly (or whatever other CDNs we decide to use in the future).

Comment (dwradcliffe) [inline app/views/pages/download.html.erb]: That's a good point, since this is about to change again.

Comment (sferik) [inline app/views/pages/download.html.erb]: This is something we have to think about a lot at SoundCloud, where we dynamically switch between CDNs for streaming audio based on factors such as price, availability, and requesting IP address (e.g. in our testing, some CDNs have dramatically better performance to Australia and parts of Asia). Having a canonical URL gives us a lot more flexibility during a CDN outage (or when a particular CDN’s IPs are block by an entire country). In our experience, this flexibility is worth paying the price of making an extra hop.

Comment (dwradcliffe) [inline app/views/pages/download.html.erb]: @reedloden Can you replace these domains with `https://rubygems.org/`. I've added redirects to our current CDN links.

Comment (reedloden) [inline app/views/pages/download.html.erb]: @dwradcliffe done!

**Comment by reedloden** [comment]
@dwradcliffe, any luck getting this reviewed and merged? Would like to stop having to rebase every so often. :)

**Comment by dwradcliffe** [comment]
:+1:

**Pull request merged**